packages/gatsby/src/utils/babel-loader.js

```diff
@@ -0,0 +1,71 @@
+const babelLoader = require(`babel-loader`)
+const path = require(`path`)
+const fs = require(`fs`)
+
+const prepareOptions = babel => {
+  const pluginBabelConfig = require(path.join(
+    process.cwd(),
+    `./.cache/babelState.json`
+  ))
+
+  const stage = fs.readFileSync(
+    path.join(process.cwd(), `./.cache/current-stage.txt`),
+    `utf8`
+  )
+
+  // Go through babel state and create config items.
+  const reduxPlugins = []
+  const reduxPresets = []
+  pluginBabelConfig[stage].plugins.forEach(plugin => {
+    reduxPlugins.push(
+      babel.createConfigItem([require.resolve(plugin.name), plugin.options], {
```

Comment:
Might be worth caching this somehow to avoid re-resolving for every single file that Gatsby compiles.

Reply:
Yeah, node caches that so we should be fine.